R/new.R

```diff
@@ -4,11 +4,20 @@
 #'
 #' @param x A tibble-like object
 #' @param ... Passed on to [structure()]
+#' @param nrow The number of rows, guessed from the data by default
 #' @param subclass Subclasses to assign to the new object, default: none
 #' @export
-new_tibble <- function(x, ..., subclass = NULL) {
+new_tibble <- function(x, ..., nrow = NULL, subclass = NULL) {
+  stopifnot(is.list(x))
+  stopifnot(has_nonnull_names(x))
+
   x <- update_tibble_attrs(x, ...)
   x <- set_tibble_class(x, subclass = subclass)
+  # Make sure that we override any row names that
+  # may have been there previously, in x or in ...
+  if (is.null(nrow)) nrow <- guess_nrow(x)
```

Comment:
The comment doesn't seem right to me - I think it should just be "set row names" (in which case you can just remove it)